skimage/restoration/inpaint.py

```diff
@@ -126,25 +252,88 @@ def inpaint_biharmonic(image, mask, multichannel=False):
     image = skimage.img_as_float(image)
     mask = mask.astype(bool)
 
-    # Split inpainting mask into independent regions
-    kernel = ndi.morphology.generate_binary_structure(mask.ndim, 1)
-    mask_dilated = ndi.morphology.binary_dilation(mask, structure=kernel)
-    mask_labeled, num_labels = label(mask_dilated, return_num=True)
-    mask_labeled *= mask
-
     if not multichannel:
         image = image[..., np.newaxis]
-
     out = np.copy(image)
 
-    for idx_channel in range(image.shape[-1]):
-        known_points = image[..., idx_channel][~mask]
-        limits = (np.min(known_points), np.max(known_points))
-
-        for idx_region in range(1, num_labels+1):
-            mask_region = mask_labeled == idx_region
-            _inpaint_biharmonic_single_channel(mask_region,
-                out[..., idx_channel], limits)
+    # Create biharmonic coefficients ndarray
+    radius = 2
+    coef_shape = (2 * radius + 1,) * mask.ndim
+    coef_center = (radius,) * mask.ndim
+    neigh_coef_full, coef_idx, coef_vals = _get_neigh_coef(coef_shape,
+                                                           coef_center,
+                                                           coef_cache={},
+                                                           dtype=out.dtype)
+
+    # stride for the last spatial dimension
+    channel_stride_bytes = out.strides[-2]
+
+    # offsets to all neighboring non-zero elements in the footprint
+    offsets = tuple(c - radius for c in coef_idx)
```

Comment:
Isn't `coef_idx` a `numpy.array`?
```suggestion
    offsets = coef_idx - radius
```